shared/src/main/java/uk/gov/di/authentication/shared/services/AuthenticationAttemptsService.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.di.authentication.shared.services;
+
+import uk.gov.di.authentication.shared.entity.AuthenticationAttempts;
+import uk.gov.di.authentication.shared.helpers.NowHelper;
+
+import java.util.Optional;
+
+public class AuthenticationAttemptsService extends BaseDynamoService<AuthenticationAttempts> {
+
+    public AuthenticationAttemptsService(ConfigurationService configurationService) {
+        super(AuthenticationAttempts.class, "authentication-attempt", configurationService);
+    }
+
+    public void createOrIncrementCount(
+            String internalSubjectId, long ttl, String journeyType, String countType) {
+        Optional<AuthenticationAttempts> authenticationAttempt =
+                get(internalSubjectId, buildSortKey(journeyType, countType));
+        if (authenticationAttempt.isPresent()) {
+            authenticationAttempt.get().setCount(authenticationAttempt.get().getCount() + 1);
+            authenticationAttempt.get().setTimeToLive(ttl);
+        } else {
+            authenticationAttempt =
+                    Optional.ofNullable(
+                            new AuthenticationAttempts()
+                                    .withInternalSubjectId(internalSubjectId)
+                                    .withCount(1)
+                                    .withCountType(countType)
+                                    .withJourneyType(journeyType)
+                                    .withTimeToLive(ttl));
+        }
+        authenticationAttempt.ifPresent(this::update);
+    }
+
+    public int getAuthenticationAttempt(
+            String internalSubjectId, String journeyType, String countType) {
+        long currentTimestamp = NowHelper.now().toInstant().getEpochSecond();
+        var record =
+                get(internalSubjectId, buildSortKey(journeyType, countType))
+                        .filter(t -> t.getTimeToLive() > currentTimestamp);
+        if (record.isEmpty()) return -1;
+        return record.get().getCount();
+    }
+
+    public void deleteCount(String internalSubjectId, String journeyType, String countType) {
```

Comment:
This method doesn't delete a count it sets it to zero.  I think the better approach is to delete the Item.